### PR TITLE
feat: setup release branch cut github action

### PR DESCRIPTION
**Pull request by rajat43**
# Description

feat: setup release branch cut github action

- closes 0xsquid/squid-core# (issue number)
- Zenhub issue: (optional)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (code improvement)
- [ ] Styles (adding, editing or fixing styles)
- [ ] Unit test

# How Has This Been Tested?

Please describe the steps to test.

## Evidence

Please add some evidence like screenshots or records.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,5 +66,26 @@
...
+      "release": true,
+      "tokenRef": "GITHUB_AUTH"
+    },
+    "git": {
+      "tagName": "v${version}",
+      "pushRepo": "https://github.com/0xsquid/squid-core.git",
+      "commitMessage": "chore: branch cut release for api v${version}",
+      "requireBranch": "release/*"
+    },
+    "npm": {
```
Comment (0x1NotMe) [inline package.json:87]: We probably want to publish to NPM

**Review by 0x1NotMe — COMMENTED** [review]
We should also remove the release stage from the other workflow

File: package.json (hunk 1 above)
Comment (rajat43) [reply package.json:87]: Added NPM publish step to the script

File: package.json [hunk 2]
```diff
@@ -60,5 +66,26 @@
...
+        "infile": "CHANGELOG.md"
+      }
+    },
+    "github": {
+      "release": true,
+      "tokenRef": "GITHUB_AUTH"
+    },
+    "git": {
+      "tagName": "v${version}",
+      "pushRepo": "https://github.com/0xsquid/squid-core.git",
```
Comment (0x1NotMe) [inline package.json:83]: Should be sdk repo

File: .github/workflows/release-cut.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        description: Release type (major/minor/patch/prerelease)
+        required: true
+      branch-name:
+        description: Branch name (Optional), defaults to main
+        required: false
+        default: main
+
+jobs:
+  cut-release:
+    if: contains('["0x1NotMe","0x0Koda", "rajat43"]', github.actor)
```
Comment (0x1NotMe) [inline .github/workflows/release-cut.yml]: Please add Juan for release people

File: package.json (hunk 2 above)
Comment (rajat43) [reply package.json]: updated now

**Review by 0x1NotMe — APPROVED** [review]

File: .github/workflows/release-cut.yml (hunk 3 above)
Comment (rajat43) [reply .github/workflows/release-cut.yml]: done

**Pull request merged**